gamemodes/terrortown/entities/entities/ttt_flame.lua

```diff
@@ -189,73 +122,81 @@ if SERVER then
-					return
-				end
-
 				-- deal damage
 				local dmg = DamageInfo()
 				dmg:SetDamageType(DMG_BURN)
-				dmg:SetDamage(math.random(4,6))
+				dmg:SetDamage(self.hurt_base + math.random(-self.hurt_variance, self.hurt_variance))
 				if IsValid(self:GetDamageParent()) then
 					dmg:SetAttacker(self:GetDamageParent())
 				else
 					dmg:SetAttacker(self)
 				end
 				dmg:SetInflictor(self.firechild)
 
-				RadiusDamage(dmg, self:GetPos(), 132, self)
+				gameEffects.RadiusDamage(dmg, self:GetPos(), self.hurt_radius, self)
 
 				self.next_hurt = CurTime() + self.hurt_interval
 			end
 
 			return
-		elseif self:GetVelocity() == Vector(0,0,0) then
-			if self:WaterLevel() > 0 then
+		else
+			if self:GetBurning() then
+				-- we already were lit once, but now our child is missing
+				-- we should just die, instead
 				self:SetDieTime(0)
-
-				return
+			else
+				-- wait until we're still before creating a fire
+				if self:GetVelocity() == Vector(0,0,0) then
+					self:StartFire()
+				end
 			end
-
-			self.firechild = SpawnFire(self:GetPos(), self.fireparams.size, self.fireparams.growth, 999, self:GetDamageParent(), self)
-			self:DeleteOnRemove(self.firechild)
-
-			self:SetBurning(true)
 		end
 	end
 end
 
-if CLIENT then
-	---
-	-- @realm client
-	CreateConVar("ttt_fire_fallback", "0", FCVAR_ARCHIVE)
-
-	local fakefire = Material("cable/smoke")
-	local side = Angle(-90, 0, 0)
-
-	---
-	-- @realm client
-	function ENT:BackupDraw()
-		if not self:GetBurning() then return end
+---
+-- Begins the burning effect and activates flames.
+-- @realm shared
+function ENT:StartFire()
+	util.PaintDown(self:GetPos(), "Scorch", self)
 
-		local vstart = self:GetPos()
-		local vend = vstart + Vector(0, 0, 90)
+	self.firechild = gameEffects.SpawnFire(self:GetPos(), self.real_scale, self.fire_ignite_delay, self:GetLifeSpan(), self:GetDamageParent(), self)
+	self:DeleteOnRemove(self.firechild)
 
-		side.r = side.r + 0.1
+	self:SetBurning(true)
 
-		cam.Start3D2D(vstart, side, 1)
-		draw.DrawText("FIRE! IT BURNS!", "Default", 0, 0, COLOR_RED, TEXT_ALIGN_CENTER)
-		cam.End3D2D()
+	if self:GetImmobile() then
+		self:SetMoveType( MOVETYPE_NONE )
+		local physobj = self:GetPhysicsObject()
+		physobj:EnableMotion(false)
+	end
+end
 
-		render.SetMaterial(fakefire)
-		render.DrawBeam(vstart, vend, 80, 0, 0, COLOR_RED)
+if CLIENT then
+	local flamesprites = {}
+	for i = 1, 5, 1 do
+		flamesprites[#flamesprites + 1] = Material("particles/flamelet" .. i)
 	end
 
 	---
 	-- @realm client
 	function ENT:Draw()
+		if self:GetBurning() and self.debug_viz then
+			render.DrawWireframeSphere(self:GetPos(), self.hurt_radius, 16, 9, COLOR_RED)
+		elseif not self:GetBurning() then
+			local frame = math.floor(((SysTime() * self.low_detail_fps) + self:EntIndex()) % #flamesprites) + 1
+			cam.Start3D()
+			render.SetMaterial(flamesprites[ frame ])
+			render.DrawSprite(self:GetPos(), self.low_detail_size, self.low_detail_size, color_white)
+			cam.End3D()
+		end
```

Comment:
What is this exactly? 
This was not in here before and i don't know why this entity is drawing anything at all in this state?

Reply:
The entity is not burning because it has not gone from invisible to visible yet, so to give it a material to track it, it is drawing a billboarded sprite at its center of mass. GetBurning is poorly named but it was already like that.